demos/portal-embed-demo/README.md

```diff
@@ -0,0 +1,148 @@
+
+For now, Portal will not process user input. If you want to interact with the Portal (like playing the audio in the demo), use `postMessage`/`onmessage`.
+```javascript
+// Send message from the portal element
+const portal = document.querySelector('portal');
+portal.postMessage({someKey: someValue}, ORIGIN);
+
+// Receive message via window.portalHost
+window.portalHost.addEventListener('message', evt => {
+   const data = evt.data;
+   // do the actual following
+});
+```
+> Demo code reference: interacting with the audio player ([sending messages](https://github.com/WICG/portals/blob/master/demos/portal-embed-demo/public/js/portalog/portals-controller.js#L135) and [receiving messages](https://github.com/WICG/portals/blob/master/demos/portal-embed-demo/public/js/ttt/portals-controller.js#L183))
+
+When the user decides to navigate the the Portal content i.e. click, it is a good opportunity to animate the Portal and then call the `activate` function. User will be navigated to the Portal content seamlessly (but the URL changes). You can also see that the session of the Portal content is consistent and the audio keeps playing even after this activation.
+```javascript
+// do some fancy animations and..
+// activate portal
+const portal = document.querySelector('portal');
+portal.activate();
+```
+> Demo code reference: [doing some animation on user click](https://github.com/WICG/portals/blob/master/demos/portal-embed-demo/public/js/portalog/portals-controller.js#L47) and [activating the Portal](https://github.com/WICG/portals/blob/master/demos/portal-embed-demo/public/js/portalog/portals-controller.js#L85) (note that you can optionally pass [custom data](https://github.com/WICG/portals/blob/master/demos/portal-embed-demo/public/js/portalog/portals-controller.js#L86) to the Portal)
+
+For the Portal content side, you can listen to the `portalactivate` event to check if the page was activated. Also from the event you can retrieve the previous page as a Portal element by calling the `adoptPredecessor` function. By leveraging the predecessor Portal element, you can implement a seamless navigation experience when going back and forth between the two pages (or origins).
+```javascript
+// Listen to the portalactivate event
+window.addEventListener('portalactivate', evt => {
+   // ... and creatively use the predecessor
+   const portal = evt.adoptPredecessor();
+   document.querySelector('someElm').appendChild(portal);
+});
+```
+> Demo code reference: [listening to `portalactivate`](https://github.com/WICG/portals/blob/master/demos/portal-embed-demo/public/js/ttt/portals-controller.js#L144) and [reusing the predecessor](https://github.com/WICG/portals/blob/master/demos/portal-embed-demo/public/js/ttt/portals-controller.js#L152)
+
+On the predcessor side, `activate` will resolve with an `undefined` `Promise` when the Portal activation has completed. If it was adopted as a predecessor, you can start using `window.portalHost`.
```

Comment:
```suggestion
On the predecessor side, `activate` will resolve with an `undefined` `Promise` when the Portal activation has completed. If it was adopted as a predecessor, you can start using `window.portalHost`.
```

Up to you, but I'm also not sure it's interesting to point out that a `Promise<void>` resolves with `undefined`. Maybe just say "`activate` returns a promise that resolves when activation has completed."?

Reply:
Thank you for the comment! I wasn't so used to using void Promises in other cases so was adding this comment but as you mentioned since it's <void> it's obvious it will not return something. Changed the comments.
https://github.com/WICG/portals/pull/105/commits/cfabda684eec194839e14ed66a384d869d641852